ql/src/java/org/apache/hadoop/hive/ql/metadata/Hive.java

```diff
@@ -2171,22 +2172,32 @@ private Materialization getMaterializationInvalidationInfo(MaterializedViewMetad
         materialization.setSourceTablesCompacted(true);
         return materialization;
       }
-      Boolean b = storageHandler.hasAppendsOnly(
-          table, mvSnapshot.getTableSnapshots().get(table.getFullyQualifiedName()));
-      if (b == null) {
+
+      boolean hasAppendsOnly = true;
+      boolean noSnapshots = true;
+      for (SnapshotContext snapshot : storageHandler.getSnapshots(
```

Comment:
should we check if snapshots are supported?
````
mvSnapshot.getTableSnapshots().get(table.getFullyQualifiedName())
````
won't create any issues without check?

Reply:
Source table types are checked at MV creation. Also the default implementation of getSnapshots returns empty list which means we have to fall back to full rebuild. This is also the case when Iceberg table is empty so there are no snapshots. But I added a check It doesn't hurt.